services/users/src/main/java/org/opfab/users/controllers/UsersController.java

```diff
@@ -164,7 +171,8 @@ public UserSettings fetchUserSetting(HttpServletRequest request, HttpServletResp
     public UserSettings patchUserSettings(HttpServletRequest request, HttpServletResponse response,
             @PathVariable("login") String login,
             @Valid @RequestBody UserSettings userSettings) throws ApiErrorException {
-        OperationResult<UserSettings> result = userSettingsService.patchUserSettings(login, userSettings);
+        User user = this.extractUserFromJwtToken(request);
+        OperationResult<UserSettings> result = userSettingsService.patchUserSettings(user, login, userSettings);
```

Comment:
> _⚠️ Potential issue_
> 
> **Ensure User Consistency Between JWT and Path Parameter**
> 
> There is a potential security issue: the user extracted from the JWT token may not match the `login` provided in the path parameter. This could allow a user to modify another user's settings.
> 
> Consider adding a check to ensure that the user from the JWT token matches the `login` path parameter to prevent unauthorized access.
> 
> Apply this diff to add the validation:
> 
> ```diff
>  public UserSettings patchUserSettings(HttpServletRequest request, HttpServletResponse response,
>          @PathVariable("login") String login,
>          @Valid @RequestBody UserSettings userSettings) throws ApiErrorException {
>      User user = this.extractUserFromJwtToken(request);
> +    if (!user.getLogin().equalsIgnoreCase(login)) {
> +        throw buildApiException(HttpStatus.BAD_REQUEST, NO_MATCHING_USER_NAME_MSG);
> +    }
>      OperationResult<UserSettings> result = userSettingsService.patchUserSettings(user, login, userSettings);
>      if (result.isSuccess())
>          return result.getResult();
>      else
>          throw createExceptionFromOperationResult(result);
>  }
> ```
> 
> 📝 Committable suggestion

@coderabbitai  This is done before in a specific security spring class